### ISSUE TITLE
Unknown long option: "output-count-fn"

### ISSUE THREAD

**Issue by a-velt** [PacificBiosciences/pbbioconda#451]
Hi,

**Operating system**
CentOS Linux 7

**Package name**

For the installation of pb-assembly I worked like this:

Creation of a conda env with miniconda, then :
```
conda update --all
conda install pb-assembly
```
I have encountered problems with samtools 1.7, and I found a solution in your issues which was to update to 1.9. This allowed fc_run to work

`conda install -c bioconda samtools=1.9 --force-reinstall`

I had a falcon error telling me it couldn't find pbbam, i installed it by hand and it worked.

`conda install -c bioconda pbbam`

Now, I launch a conda update --all, conda want to do this : 

```
The following packages will be UPDATED:

  curl                                    7.69.1-hbc83047_0 --> 7.78.0-h1ccaba5_0
  krb5                                    1.17.1-h173b8e3_0 --> 1.19.2-hac12032_0
  libcurl                                 7.69.1-h20c2e04_0 --> 7.78.0-h0b77cf5_0
  libedit                           3.1.20181209-hc058e9b_0 --> 3.1.20210910-h7f8727e_0
  libffi                                3.2.1-hf484d3e_1007 --> 3.3-he6710b0_2
  ncurses                                    6.1-he6710b0_1 --> 6.3-h7f8727e_2
  python                                   3.7.6-h0371630_2 --> 3.7.11-h12debd9_0
  readline                                   7.0-h7b6447c_5 --> 8.1-h27cfd23_0
  sqlite                                  3.31.1-h7b6447c_0 --> 3.36.0-hc218d9a_0

The following packages will be DOWNGRADED:

  cffi                                1.15.0-py37h7f8727e_0 --> 1.14.6-py37h400218f_0
  samtools                                  1.9-h10a08f8_12 --> 1.7-1
```

I think that these updates do not impact the error that I am going to show you and especially, if I go back to samtools 1.7 I know that I would have an error with fc_run.

**Conda environment**

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             4.5                       1_gnu
bedtools                  2.30.0               h7d7f7ad_1    bioconda
blas                      1.0                         mkl
blasr                     5.3.5                         0    bioconda
brotlipy                  0.7.0           py37h27cfd23_1003
bwa                       0.7.17               h5bf99c6_8    bioconda
bzip2                     1.0.8                h7b6447c_0
c-ares                    1.17.1               h27cfd23_0
ca-certificates           2021.10.26           h06a4308_2
certifi                   2021.10.8        py37h06a4308_0
cffi                      1.15.0           py37h7f8727e_0
charset-normalizer        2.0.4              pyhd3eb1b0_0
conda                     4.10.3           py37h06a4308_0
conda-package-handling    1.7.3            py37h27cfd23_1
cryptography              35.0.0           py37hd23ed53_0
curl                      7.69.1               hbc83047_0
falcon-kit                1.8.1                    pypi_0    pypi
falcon-phase              1.2.0                    pypi_0    pypi
falcon-unzip              1.3.7                    pypi_0    pypi
future                    0.18.2                   py37_1
htslib                    1.9                  ha228f0b_7    bioconda
idna                      3.3                pyhd3eb1b0_0
intel-openmp              2021.4.0          h06a4308_3561
k8                        0.2.5                h9a82719_1    bioconda
krb5                      1.17.1               h173b8e3_0
ld_impl_linux-64          2.35.1               h7274673_9
libcurl                   7.69.1               h20c2e04_0
libdeflate                1.0                  h14c3975_1    bioconda
libedit                   3.1.20181209         hc058e9b_0
libev                     4.33                 h7f8727e_1
libffi                    3.2.1             hf484d3e_1007
libgcc                    7.2.0                h69d50b8_2
libgcc-ng                 9.3.0               h5101ec6_17
libgomp                   9.3.0               h5101ec6_17
libnghttp2                1.46.0               hce63b2e_0
libssh2                   1.9.0                h1ba5d50_1
libstdcxx-ng              9.3.0               hd4cf53a_17
minimap2                  2.21                 h5bf99c6_0    bioconda
mkl                       2021.4.0           h06a4308_640
mkl-service               2.4.0            py37h7f8727e_0
mkl_fft                   1.3.1            py37hd3c417c_0
mkl_random                1.2.2            py37h51133e4_0
mummer4                   4.0.0rc1        pl5262h1b792b2_1    bioconda
ncurses                   6.1                  he6710b0_1
networkx                  2.6.3              pyhd3eb1b0_0
nim-falcon                3.0.1                h1341992_0    bioconda
numpy                     1.21.2           py37h20f2e39_0
numpy-base                1.21.2           py37h79a1101_0
openssl                   1.1.1l               h7f8727e_0
pb-assembly               0.0.8                hdfd78af_1    bioconda
pb-dazzler                0.0.1                h779adbc_1    bioconda
pb-falcon                 2.2.4            py37hed50d52_0    bioconda
pb-falcon-phase           0.1.0                h3889886_0    bioconda
pbbam                     1.3.0                h8e3dc82_0    bioconda
pbgcpp                    2.0.2                h9ee0642_1    bioconda
pbmm2                     1.3.0                h56fc30b_0    bioconda
pcre                      8.45                 h295c915_0
perl                      5.26.2               h14c3975_0
pip                       21.2.2           py37h06a4308_0
pycosat                   0.6.3            py37h27cfd23_0
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 21.0.0             pyhd3eb1b0_1
pysam                     0.15.3           py37hda2845c_1    bioconda
pysocks                   1.7.1                    py37_1
python                    3.7.6                h0371630_2
python-edlib              1.3.8.post1      py37hc9558a2_0    bioconda
python-intervaltree       3.1.0              pyh864c0ab_0    bioconda
python-msgpack            0.6.1            py37hc9558a2_1    bioconda
python-sortedcontainers   2.4.0              pyh5e36f6f_0    bioconda
racon                     1.4.20               h9a82719_1    bioconda
readline                  7.0                  h7b6447c_5
requests                  2.26.0             pyhd3eb1b0_0
ruamel_yaml               0.15.100         py37h27cfd23_0
samtools                  1.9                 h10a08f8_12    bioconda
setuptools                58.0.4           py37h06a4308_0
six                       1.16.0             pyhd3eb1b0_0
sqlite                    3.31.1               h7b6447c_0
tk                        8.6.11               h1ccaba5_0
tqdm                      4.62.3             pyhd3eb1b0_1
urllib3                   1.26.7             pyhd3eb1b0_0
wheel                     0.37.0             pyhd3eb1b0_1
xz                        5.2.5                h7b6447c_0
yaml                      0.2.5                h7b6447c_0
zlib                      1.2.11               h7b6447c_3
```

**Describe the bug**

fc_run worked. Now I'm trying to run the falcon-unzip step. The falcon-unzip step finished well, then during the 4-polish step, I have this error, for example in the folder "4-polish/temp-unphased/000661F" :

```
module () {  eval `/usr/bin/modulecmd bash $*`
}

IFS=$'
        '
+ IFS='
        '
set -vxeuo pipefail
+ set -vxeuo pipefail
hostname
+ hostname
genosmp03
pwd
+ pwd
/data/avelt/2021_assemblage_Arinto/FALCON_run/4-polish/temp-unphased/000661F
date
+ date
Thu Dec  2 08:59:51 CET 2021
# Substitution will be similar to snakemake "shell".

python3 -m falcon_unzip.mains.polish_unphased_readmapping --ctg 000661F --fai ../../input/combined_ph.fasta.fai --out-read-names read_names.txt --out-ref-names ref_names.txt --read-to-ctg ../../input/read2ctg.txt
+ python3 -m falcon_unzip.mains.polish_unphased_readmapping --ctg 000661F --fai ../../input/combined_ph.fasta.fai --out-read-names read_names.txt --out-ref-names ref_names.txt --read-to-ctg ../../input/read2ctg.txt
falcon-unzip 1.3.7 (pip thinks "falcon-unzip 1.3.7")
samtools fqidx -r read_names.txt ../../input/preamble.fastq > reads.fastq
+ samtools fqidx -r read_names.txt ../../input/preamble.fastq
samtools faidx -r ref_names.txt ../../input/combined_ph.fasta  > ref.fasta
+ samtools faidx -r ref_names.txt ../../input/combined_ph.fasta
pbmm2 align -j 1 --preset CCS --sort ref.fasta reads.fastq > tmp.bam
+ pbmm2 align -j 1 --preset CCS --sort ref.fasta reads.fastq
>|> 20211202 07:59:52.935 -|- WARN -|- CheckPositionalArgs -|- 0x7f2108f0c740|| -|- Input is FASTQ. Output BAM file cannot be used for polishing with GenomicConsensus!
falconc bam-filter-clipped -t -F 0x704 --output-count-fn filtered_aln_count.txt --input-fn:- --output-fn:- < tmp.bam | samtools view -q 5 -bh > aln.bam
+ falconc bam-filter-clipped -t -F 0x704 --output-count-fn filtered_aln_count.txt --input-fn:- --output-fn:-
+ samtools view -q 5 -bh
  version=0.2.1+git.fatal: Not a git repository (or any of the parent directories): .git, nim-version=1.0.2
Unknown long option: "output-count-fn"

Run with --help for full usage.
```

It seems that I am using the wrong version of falconc since it cannot find the option "output-count-fn". But I don't see falconc in my conda environment. Which tool does it belong to and which version should I use?

**Error message**

See abvove

**To Reproduce**

Here is my fc_unzip.cfg : 
```

[General]
max_n_open_files = 900000

[Unzip]
fastq=/data/avelt/2021_assemblage_Arinto/rawdata/ccs_Arinto-AR36.fastq
input_fofn=ccs.fasta.fofn
input_bam_fofn=ccs.bam.fofn

[job.defaults]
JOB_QUEUE = bigmem
MB = 50000
NPROC = 12
job_type=local
pwatcher_type=blocking
submit = bash -C ${CMD} >| ${STDOUT_FILE} 2>| ${STDERR_FILE}

[job.high]
njobs=1
NPROC=24
MB=25000

[job.highmem]
njobs=10
MB=60000
(base) avelt@
```

It's difficult to give a subset of the dataset, what exactly do you want?

**Expected behavior**

I expect the "4-polish" step to finish correctly.

Thank you very much for your help,
Amandine

**Comment by bingochenbin** [comment]
same question too

**Comment by a-velt** [comment]
Hi,

Up ?

**Comment by EricRLucas** [comment]
I'm also having this issue (Linux Mint 20.2)

**Comment by isovic** [comment]
Hi,

`falconc` is a tool used by our assembly workflows, consisting of a number of subtools.
It appears that you somehow have an old version installed.
Could you run `which falconc` and `falconc version` to see where it comes from and which version you have installed?
Further, can you try installing the `pb-falconc` module manually from Bioconda? The current version is `1.15.0`.
It should be available from version >= `1.2.0`.

Best regards,
Ivan.
